elpy.el

```diff
@@ -2050,14 +2064,17 @@ This requires Python 2.7 or later."
 (put 'elpy-test-discover-runner 'elpy-test-runner-p t)
 
 (defun elpy-test-django-runner (top file module test)
-  "Test the project using the Django discover runner.
+  "Test the project using the Django discover runner,
+   or with manage.py if elpy-test-django-with-manage is true.
 
 This requires Django 1.6 or the django-discover-runner package."
   (interactive (elpy-test-at-point))
   (if module
       (apply #'elpy-test-run
              top
-             (append elpy-test-django-runner-command
+             (append (if elpy-test-django-with-manage
+                         elpy-test-django-runner-manage-command
+                       elpy-test-django-runner-command)
                      (list (if test
```

Comment:
`(concat (locate-dominating-file (buffer-file-name) "manage.py") "manage.py")` will give you the full path to a `manage.py` which you can use as the command to run. This is a lot cleaner than your solution, as it will work in a lot more situations. Please also think about what "the command to run" means, as that will simplify the rest of your code.